helper/helper.go

```diff
@@ -363,3 +366,27 @@ func TLScertToFile(filename string, derBytes []byte) error {
 
 	return err
 }
+
+// FormatUploadFilePath ensures that path separators are "/", and returns error if the
+// filepath contains a disallowed character matched with regex
+func FormatUploadFilePath(filePath string) (string, error) {
+
+	// Check for mixed "\" and "/" in filepath. Stop and throw an error if true so that
+	// we do not end up with unintended folder structure when applying ToSlash later on
+	if strings.Contains(filePath, "\\") && strings.Contains(filePath, "/") {
+		return filePath, fmt.Errorf("filepath contains mixed '\\' and '/' characters")
+	}
+
+	// make path separators os compatible
+	outPath := filepath.ToSlash(filePath)
```

Comment:
Given a filepath `path1\path2\path2` (coming from a client running in windows), the above line should change it to `path1/path2/path3` which is the desired behavior.

Reply:
It will not do that since the S3proxy is *not* running on windows.

```go
// ToSlash returns the result of replacing each separator character
// in path with a slash ('/') character. Multiple separators are
// replaced by multiple slashes.
func ToSlash(path string) string {
	if Separator == '/' {
		return path
	}
	return strings.ReplaceAll(path, string(Separator), "/")
}
```
`Separator` is the `os.PathSeparator` so on a linux machine it will just return the string as it is.

Reply:
yep, I got stuck in a loop. Now I see how running the unitests in windows was a bad idea! I now also get the comments on the unitests below. I will fix this.